L1Trigger/L1THGCal/src/veryfrontend/HGCalVFECompressionImpl.cc

```diff
@@ -1,5 +1,87 @@
 #include "L1Trigger/L1THGCal/interface/veryfrontend/HGCalVFECompressionImpl.h"
 
 HGCalVFECompressionImpl::
-HGCalVFECompressionImpl(const edm::ParameterSet& conf)
-{}
+HGCalVFECompressionImpl(const edm::ParameterSet& conf):
+    exponentBits_(conf.getParameter<uint32_t>("exponentBits")),
+    mantissaBits_(conf.getParameter<uint32_t>("mantissaBits")),
+    rounding_(conf.getParameter<bool>("rounding"))
+{
+  saturable_ = false;
+  saturationValue_ = 0xffffffff;
+  if (((1 << exponentBits_) + mantissaBits_ - 1) < 32) {
+    saturable_ = true;
+    saturationValue_ = (1 << ((1 << exponentBits_) + mantissaBits_ - 1)) - 1;
+  }
+  // TODO: for non-saturating need to get maximum code point as well?
+  // i.e. 8-bit compressed code that would hit 0xffffffff
+
+  // build the compressed values lookup table
```

Comment:
See my other comments: I think it would be clearer if the inflated value is computed on the fly, rather than using a pre-built LUT